app/controllers/stories/feeds_controller.rb

```diff
@@ -0,0 +1,21 @@
+class Stories::FeedsController < ApplicationController
+  respond_to :json
+
+  def show
+    @stories = assign_feed_stories
+  end
+
+  private
+
+  def assign_feed_stories
+    feed = Articles::Feed.new(number_of_articles: 35, page: @page, tag: params[:tag])
+    stories = if %w[week month year infinity].include?(params[:timeframe])
```

Comment:
@citizen428 Well, I'm about to instantiation about forty bajillion objects with the next ActiveRecord call, but your point is well taken! This really does belong on the `TimeFramer` object IMHO, will move!

**LOVE** your suggestions BTW: they make me crabby only because I didn't think of them first!

Reply:
@citizen428 I like that approach, but the the issue is that logic used to get stories is slightly different in the two places it's currently used is slightly different enough that encapsulating it would be weird to me. I'm going to pass on collapsing it for now: once feed handling is out of `StoriesController` it will make more sense to bring this logic into `Articles::Feed`!

Reply:
@joshpuetz Yes, on closer reading of the two code paths they are different enough that unifying them would be wrong/premature 👍